tests/FixerFactoryTest.php

```diff
@@ -387,4 +387,22 @@ private function createFixerMock($name, $priority = 0)
 
         return $fixer;
     }
+
+    /**
+     * @dataProvider provideConflictingFixersRules
+     * @expectedException \UnexpectedValueException
+     * @expectedExceptionMessageRegExp #^The rules contain conflicting fixers:\n#
+     */
+    public function testConflictingFixers(RuleSet $ruleSet)
+    {
+        FixerFactory::create()->registerBuiltInFixers()->useRuleSet($ruleSet);
+    }
+
+    public function provideConflictingFixersRules()
+    {
+        return array(
+            array(new RuleSet(array('short_array_syntax' => true, 'long_array_syntax' => true))),
+            array(new RuleSet(array('long_array_syntax' => true, 'short_array_syntax' => true))),
```

Comment:
would also be great to check if we won't have sth like:
```
The rules contain conflicting fixers:
short_array_syntax with long_array_syntax
long_array_syntax with short_array_syntax
```

Sorry if you will find it over-specified, but I have strong feeling that during implementing logic on FixerInterface side we could be catch by that.